src/visual.ts

```diff
@@ -513,42 +527,57 @@ module powerbi.extensibility.visual {
 
                 this.behavior.renderSelection(false);
             }
+        }
 
-            this.axes.attr("transform", SVGUtil.translate(this.margin.left, 0));
-            this.axisX.attr("transform", SVGUtil.translate(0, this.viewport.height - this.margin.bottom));
-            this.axisY.attr("transform", SVGUtil.translate(0, this.margin.top));
-
-            this.calculateAxes();
-            if (this.data.settings.categoryAxis.show) {
-                this.axisX.call(this.xAxisProperties.axis);
-                this.axisX.classed(StreamGraph.XAxis.className, true);
-            } else {
-                this.axisX.classed(StreamGraph.XAxis.className, false);
-                this.axisX
-                    .selectAll("*")
-                    .remove();
+        private calculateShift(): number {
+            if (!this.data.settings.categoryAxis.rotateLabels) {
+                return StreamGraph.TickHeight;
             }
-            if (this.data.settings.valueAxis.show) {
-                this.axisY.call(this.yAxisProperties.axis);
-                this.axisY.classed(StreamGraph.YAxis.className, true);
-            } else {
-                this.axisY.classed(StreamGraph.YAxis.className, false);
-                this.axisY
+
+            const rotatedTextWidth: number = StreamGraph.XAxisLabelSize * StreamGraph.Sin60,
+                rotatedLongestTextHeight: number = this.data.maxNumberOfAxisXValues * StreamGraph.Cos60;
+
+            return rotatedLongestTextHeight + rotatedTextWidth + StreamGraph.TickHeight;
+        }
+
+        private setTextNodesPosition(xAxisTextNodes: Selection<any>,
+                                     textAnchor: string,
+                                     dx: string,
+                                     dy: string,
+                                     transform: string): void {
+
+            xAxisTextNodes
+                .style("text-anchor", textAnchor)
+                .attr({ dx, dy, transform });
+        }
+
+        private toggleAxisVisibility(
+            isShow: boolean,
+            axisType: string): void {
+
+            this["axis" + axisType].classed(StreamGraph[axisType + "Axis"].class, isShow);
```

Comment:
It'd be better to use an object directly and specify it in the arguments of the method instead.
```javascript
toggleAxisVisibility(isShown: boolean, axis: d3.svg.axis) {
  axis.classed(...
```